TelemetryManager/Src/TMCircularBuffer.cpp

```diff
@@ -1,30 +1,56 @@
 
-#include "CircularBuffer.hpp"
+#include "TMCircularBuffer.hpp"
 
-TMCircularBuffer::TMCircularBuffer(uint8_t* buf, uint16_t size): CircularBuffer(buf, size) {
+TMCircularBuffer::TMCircularBuffer(uint8_t* buf, uint16_t size): CircularBuffer(*buf, size) {
     // Constructor
```

Comment:
Things about cpp inheritance, and the way how you inherit from CircularBuffer will not make your TMCircularBuffer work. 
If your class tries to inherit the private variables from the superclass, you won't get access to it. And I think a lot of methods depend on those private variables. Should change the parent class private variables to protected to allow your class to keep using them.
https://www.geeksforgeeks.org/inheritance-in-c/